tests/formats/format_helpers.test.ts

```diff
@@ -338,6 +386,15 @@ describe("formatValue on number", () => {
     expect(formatValue(1234, { format: "#,##0[$TEST][$TEST]", locale })).toBe("1,234TESTTEST");
     expect(formatValue(1234, { format: "[$TEST][$TEST]#,##0", locale })).toBe("TESTTEST1,234");
   });
+
+  test("Cannot mix text, date and number formats", () => {
+    expect(() => formatValue(1234, { format: "#,##0 dd", locale })).toThrow();
+    expect(() => formatValue(1234, { format: "0 mm", locale })).toThrow();
+    expect(() => formatValue(1234, { format: "# yyyy", locale })).toThrow();
+    expect(() => formatValue(1234, { format: "@ dd", locale })).toThrow();
+    expect(() => formatValue(1234, { format: "# @", locale })).toThrow();
+    expect(() => formatValue(1234, { format: "@0", locale })).toThrow();
```

Comment:
it seems that this is not the case when it comes to the text part :
ex: `";;;@0"`

Reply:
What do you mean ? The format `;;;@0` does not work in Excel. If you're testing on GSheet, I'm 99% sur it automatically escapes strings, so the format will be converted to somethign like `;;;@"0"`